src/persist/disk.rs

```diff
@@ -19,6 +20,10 @@ pub struct DiskBackend {
     index: Index<Id, u64>,
     data_path: PathBuf,
     data_ofs: u64,
+    // in the case of an ephemeral store, we need to extend the lifetime of the
+    // `TempDir` by storing it in the struct
+    #[allow(unused)]
```

Comment:
Seems to me unnecessary the `#[allow(unused)]`, since it always used (you set it to `None` in `new`)

Reply:
set yes, but it's never read so it throws a warning.